src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -291,4 +301,22 @@ static class Blob {
       this.blobStats = blobStats;
     }
   }
+
+  /**
+   * Get the metadata size (footer size) from a parquet file
+   *
+   * @param bytes the serialized parquet file
+   * @return the extended metadata size
+   */
+  static long getExtendedMetadataSize(byte[] bytes) throws IOException {
+    final int magicOffset = bytes.length - ParquetFileWriter.MAGIC.length;
+    final int footerSizeOffset = magicOffset - Integer.BYTES;
+    if (footerSizeOffset < 0
+        || !ParquetFileWriter.MAGIC_STR.equals(
+            new String(bytes, magicOffset, ParquetFileWriter.MAGIC.length))) {
+      throw new IllegalArgumentException("Invalid parquet file");
```

Comment:
also add the various ingredients to the formulas here into the error message. bytes.length, MAGIC.length, footerSizeOffset, new String(bytes, magicOffset, ParquetFileWriter.MAGIC.length), ParquetFileWriter.MAGIC_STR.

Reply:
Added more error message.